crates/sui-storage/src/event_store/sql.rs

```diff
@@ -368,11 +383,11 @@ impl EventStore for SqlEventStore {
         // TODO: See https://kerkour.com/high-performance-rust-with-sqlite
         for event in events {
             // Skip events that have a lower sequence number... which must be same or increasing
-            if event.seq_num < cur_seq {
-                debug!(tx_digest =? event.tx_digest, seq_num = event.seq_num, cur_seq, "Skipping event with lower sequence number than current");
+            if event.tx_seq_num < cur_seq {
```

Comment:
So this is supposed to prevent the same events from being ingested, but it would be better to use an externally assigned sequence number for each event here, instead of tx_seq_num (which repeats for the same events in the same TX).  I know you didn't write this code originally but just saying we should improve this code if possible.